### ISSUE TITLE
IntelliSence matches the argument list

### ISSUE THREAD

**Issue by GuLicheng** [microsoft/vscode-cpptools#6924]
```
#include <iostream>
#include <vector>

struct foo
{
    using value_type = int;

    template <typename Container>
    Container to() const
    {
        return Container();
    }

    template <template <typename...> typename Container>
    Container<value_type> to() const
    {
        return Container<value_type>();
    }

};

int main()
{
    auto vec1 = foo().to<std::vector>(); // no instance of overloaded function "foo::to" matches the argument list, C/C++304
    auto vec2 = foo().to<std::vector<int>>();
  // it can built successfully
}
```
OS: windows10
extend version:1.2.0
cpp_standard: c++20

**Comment by Colengms** [comment]
Hi @GuLicheng .  Could you provide the configuration you are using from your `c_cpp_properties.json`, as well as any `C_Cpp.default.*` settings that could be influencing the configuration?

I'm not able to easily repro, using the code you posted and the following configuration:

```
        {
            "name": "cl.exe",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "compilerPath": "cl.exe",
            "cppStandard": "c++20"
        },
```

Also, could you run the `C/C++: Log Diagnostics` command when this repro's, and provide that output?  That would tell us exactly how IntelliSense is getting configured for that file.

**Comment by michelleangela** [comment]
@GuLicheng 

This is what I get when I try to reproduce with your sample code:
![image](https://user-images.githubusercontent.com/38734287/107292668-13951580-6a1f-11eb-9ee7-88a82ccbd031.png)

![image](https://user-images.githubusercontent.com/38734287/107292692-1a238d00-6a1f-11eb-82e9-cdc20f56a4c6.png)

![image](https://user-images.githubusercontent.com/38734287/107292703-1ee84100-6a1f-11eb-84ac-8923cac0f88f.png)

Could you clarify what you mean by: "no instance of overloaded function "foo::to" matches the argument list"?

**Comment by GuLicheng** [comment]
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "microsoft/vscode-cpptools"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2021年2月9日(星期二) 上午7:06
收件人:&nbsp;"microsoft/vscode-cpptools"<vscode-cpptools@noreply.github.com&gt;;
抄送:&nbsp;"Yan"<576356467@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
主题:&nbsp;Re: [microsoft/vscode-cpptools] IntelliSence matches the argument list (#6924)





 
Hi @GuLicheng .  Could you provide the configuration you are using from your c_cpp_properties.json, as well as any C_Cpp.default.* settings that could be influencing the configuration?
 
I'm not able to easily repro, using the code you posted and the following configuration:
         {             "name": "cl.exe",             "includePath": [                 "${workspaceFolder}/**"             ],             "compilerPath": "cl.exe",             "cppStandard": "c++20"         },  
Also, could you run the C/C++: Log Diagnostics command when this repro's, and provide that output?  That would tell us exactly how IntelliSense is getting configured for that file.
 
—

You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

Here is my configs:

{

&nbsp; &nbsp; "configurations": [

&nbsp; &nbsp; &nbsp; &nbsp; {

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "name": "Win32",

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "includePath": [

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "${workspaceFolder}/**",

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "${CPLUS_INCLUDE_PATH}/**"

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; ],

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "defines": [

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "_DEBUG",

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "UNICODE",

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "_UNICODE"

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; ],

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "windowsSdkVersion": "10.0.18362.0",

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "compilerPath": "E:/msys2/mingw64/bin/g++.exe",

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "intelliSenseMode": "gcc-x64",

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "cStandard": "c11",

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "cppStandard": "c++20"

&nbsp; &nbsp; &nbsp; &nbsp; }

&nbsp; &nbsp; ],

&nbsp; &nbsp; "version": 4

}

I have never use&nbsp;Log Diagnostics, so I don't know how to run it.

**Comment by GuLicheng** [comment]
> @GuLicheng
> 
> This is what I get when I try to reproduce with your sample code:
> ![image](https://user-images.githubusercontent.com/38734287/107292668-13951580-6a1f-11eb-9ee7-88a82ccbd031.png)
> 
> ![image](https://user-images.githubusercontent.com/38734287/107292692-1a238d00-6a1f-11eb-82e9-cdc20f56a4c6.png)
> 
> ![image](https://user-images.githubusercontent.com/38734287/107292703-1ee84100-6a1f-11eb-84ac-8923cac0f88f.png)
> 
> Could you clarify what you mean by: "no instance of overloaded function "foo::to" matches the argument list"?


You should not use :: since it is not a static method, and the "no instance of overloaded function "foo::to" matches the argument list" is reported by intelliSence
![result](https://user-images.githubusercontent.com/52057603/107307995-c4082580-6ac2-11eb-8d10-0f4467ef8f42.png)

**Comment by Colengms** [comment]
Hi @GuLicheng .  `C/C++: Log Diagnostics` is a command that can be executed via VS Code's command palette (i.e. after pressing Control-Shift-P).

Thanks for bringing this to our attention.  I'm able to reproduce the issue on Windows when using MinGW.  I was able to reproduce something similar is VS.  (We use the same IntelliSense as VS).  I opened an issue against VS for this.  The VS code issue is here, in case you want to up-vote it: https://developercommunity.visualstudio.com/content/problem/1335274/incorrect-intellisense-no-instance-of-overloaded-f.html
